benchmarks/src/main/java/io/grpc/benchmarks/qps/Utils.java

```diff
@@ -0,0 +1,343 @@
+      context = GrpcSslContexts.forClient().trustManager(cert)
+          .sslProvider(useJdkSsl ? SslProvider.JDK : SslProvider.OPENSSL)
+          .build();
+    }
+    final EventLoopGroup group;
+    final Class<? extends io.netty.channel.Channel> channelType;
+    switch (config.transport) {
+      case NETTY_NIO: {
+        group = new NioEventLoopGroup();
+        channelType = NioSocketChannel.class;
+        break;
+      }
+      case NETTY_EPOLL: {
+        try {
+          // These classes are only available on linux.
+          Class<?> groupClass = Class.forName("io.netty.channel.epoll.EpollEventLoopGroup");
+          @SuppressWarnings("unchecked")
+          Class<? extends io.netty.channel.Channel> channelClass =
+              (Class<? extends io.netty.channel.Channel>) Class.forName(
+                  "io.netty.channel.epoll.EpollSocketChannel");
+          group = (EventLoopGroup) groupClass.newInstance();
+          channelType = channelClass;
+          break;
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+      case NETTY_UNIX_DOMAIN_SOCKET: {
+        try {
+          // These classes are only available on linux.
+          Class<?> groupClass = Class.forName("io.netty.channel.epoll.EpollEventLoopGroup");
+          @SuppressWarnings("unchecked")
+          Class<? extends io.netty.channel.Channel> channelClass =
+              (Class<? extends io.netty.channel.Channel>) Class.forName(
+                  "io.netty.channel.epoll.EpollDomainSocketChannel");
+          group = (EventLoopGroup) groupClass.newInstance();
+          channelType = channelClass;
+          break;
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+      default: {
+        // Should never get here.
+        throw new IllegalArgumentException("Unsupported transport: " + config.transport);
+      }
+    }
+    return NettyChannelBuilder
+        .forAddress(config.address)
+        .eventLoopGroup(group)
+        .channelType(channelType)
+        .negotiationType(negotiationType)
+        .executor(config.directExecutor ? MoreExecutors.newDirectExecutorService() : null)
+        .sslContext(context)
+        .connectionWindowSize(config.connectionWindow)
+        .streamWindowSize(config.streamWindow)
+        .build();
+  }
+
+  static void saveHistogram(Histogram histogram, String filename) throws IOException {
+    File file;
+    PrintStream log = null;
+    try {
+      file = new File(filename);
+      if (file.exists() && !file.delete()) {
+        System.err.println("Failed deleting previous histogram file: " + file.getAbsolutePath());
+      }
+      log = new PrintStream(new FileOutputStream(file), false);
+      histogram.outputPercentileDistribution(log, 1.0);
+    } finally {
+      if (log != null) {
+        log.close();
+      }
+    }
+  }
+
+  private static class TestServiceImpl implements TestServiceGrpc.TestService {
```

Comment:
Could we still implement this in the server? It doesn't seem good for the primary implementation of the server to be in the Util class and not in the server class.